io/include/traccc/io/binary.hpp

```diff
@@ -0,0 +1,108 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/edm/container.hpp"
+
+// VecMem include(s).
+#include <vecmem/memory/memory_resource.hpp>
+#include <vecmem/utils/copy.hpp>
+
+// System include(s).
+#include <fstream>
+
+namespace traccc {
+
+/// Function for binary file writing
+///
+/// @param out_name is the output filename which includes the path
+/// @param container is the input traccc container
+template <typename container_t>
+void write_binary(const std::string& out_name, const container_t& container) {
+    std::ofstream out_file(out_name, std::ios::out | std::ios::binary);
+
```

Comment:
Sorry, apparently Github did not like one of my comments. I am a little bit worried about marshaling data to and from disk byte-wise like this, especially because we compile this code with so many different compilers. The Itanium ABI is relatively stable and should enable us to do this in a somewhat stable fashion, but only if our data (both header and item types) are PODs. Thus I would like to suggest we add the following; 

```suggestion
    static_assert(std::is_trivial_v<container_t::header_t>, "Container header type must be trivial.");
    static_assert(std::is_standard_layout_v<container_t::header_t>, "Container header type must be standard layout.");
    static_assert(std::is_trivial_v<container_t::item_t>, "Container item type must be trivial.");
    static_assert(std::is_standard_layout_v<container_t::item_t>, "Container item type must be standard layout.");
```